frontend/components/Cluster/ClusterViewer.tsx

```diff
@@ -0,0 +1,56 @@
+import React, { useContext } from "react";
+import { swrContext } from "@/util/swr";
+import Accordion from "@mui/material/Accordion";
+import AccordionSummary from "@mui/material/AccordionSummary";
+import AccordionDetails from "@mui/material/AccordionDetails";
+import ExpandMoreIcon from "@mui/icons-material/ExpandMore";
+import { Typography, Stack, List, ListItem, Divider } from "@mui/material";
+import WindowDefinitions from "@/components/WindowFolder/WindowDefinitions";
+import { useAppSelector, useAppDispatch } from "@/util/hooks";
+import DocumentListItem from "@/components/Documents/DocumentListItem";
+import { CopyJson, CopyText, SaveDocx } from "@/components/GroupActions";
+import ButtonActions from "@/components/ButtonActions";
+
+
+export default function DocViewer(props) {
+  const swr = useContext(swrContext);
+  const { cluster } = swr.useSWRAbstract("cluster", `clusters/${props.id}`);
+  const settings = useAppSelector((state) => state.windows.settings);
+
+  const data = cluster?.history[0].included_documents.map((p) => {
+    return [p, 1.0];
+  });
+
+  const windowState = useAppSelector((state) => state.windows);
+  const wdefs = WindowDefinitions(windowState);
+
+  return (
+    <Accordion
+      defaultExpanded={settings.defaultExpanded}
+      disableGutters={true}
+      square={true}
+    >
+      <AccordionSummary
```

Comment:
just thinking, is there a way to make it so that these viewers can be generalized now? i.e. no duplicate code

Reply:
Yeah, I did consider this briefly but there are a few differences that I found made it troublesome to polymorph. I can definitely try harder if it's worthwhile. Amongst `SelectionViewer.tsx`: Cluster and Teleoscope are very similar. Notes are pretty close minus the actual data. Groups are similar but have more buttons. Docs are the most unique. In light of this, I did update the notes reader to be read-only per Ivans's request.